cobc/typeck.c

```diff
@@ -9415,6 +9432,9 @@ cb_emit_initialize (cb_tree vars, cb_tree fillinit, cb_tree value,
 		 && CB_REFERENCE_P (x)
 		 && CB_REFERENCE   (x)->subs == NULL
 		 && CB_REFERENCE   (x)->length == NULL) {
+			/* GCOS 7: Contrary to what the documentation states,
```

Comment:
You've likely have that checked in the testsuite and on GCOS - Does this apply to both the "initial state" `VAR PIC LX(5) DEPENDING ON LEN VALUE ALL 'A'` with `LEN PIC 9 VALUE 3` and with `MOVE 4 TO LEN. INITIALIZE VAR`?

Reply:
As per GCOS 7 `PIC L` fields may not have a `VALUE` clause. This is checked and rejected somewhere in this PR.

Reply:
I see, this would only leave a test like:

```cobol
MOVE 5 TO LEN
MOVE ALL 'A' TO VAR
MOVE 3 TO LEN
INITIALIZE VAR
MOVE 5 TO LEN
IF VAR = "   AA" ...
```